src/libstd/sys/redox/ext/fs.rs

```diff
@@ -204,6 +204,13 @@ pub trait MetadataExt {
     fn blocks(&self) -> u64;
 }
 
+// Hm, why are there casts here to the returned type, shouldn't the types always
+// be the same? Right you are! Turns out, however, on android at least the types
+// in the raw `stat` structure are not the same as the types being returned. Who
+// knew!
+//
+// As a result to make sure this compiles for all platforms we do the manual
+// casts and rely on manual lowering to `stat` if the raw type is desired.
```

Comment:
this change feels different than the others? If it's accurate, that's fine, but i wasn't sure.

Reply:
This comment was on the trait definition instead of the implementation (I assume that `MetadataExt` at one time had default implementations for all its methods). Sorry, I struggle with keeping the scope of my PRs limited.